### PR TITLE
tests/pkg_8g2: migrate to testrunner

### LINKED ISSUE #7871: tests: add pexpect scripts to enable `make test`
Doing the release spec tests, we found that having pexpect scripts for our tests is a good idea, as this is a first step towards automating the tests. I went now through all the tests and found, that there is a large number of them, that can easily be scripted/automated. I think it would make much sense to do so soonish, as this will save us much time down the line!

Trivial to automate:
- [x] fmt_print (#7988)
- [x] irq (#7989)
- [x] libfixmath (#8011)
- [x] libfixmath_unittests (#8011)
- [x] msg_try_receive ( #7991 )
- [x] netdev_test (#7992)
- [x] netstats_l2 (#8038)
- [x] nhdp (#7992)
- [x] pipe (#7992)
- [x] pkg_jsmn (#7992)
- [x] pkg_libcoap (#7992)
- [x] pkg_micro-ecc (#7992)
- [x] pkg_micro-ecc-with-hwrng (#7992)
- [x] pkg_minmea (#7992)
- [x] ps_schedstatistics (#7992)
- [x] pthread* , see #7906 
- [x] sched_testing (#7992)
- [x] shell (#8005)
- [x] sizeof_tcb (#7992)
- [x] ssp (#8004)
- [x] struct_tm_utility (#8008)
- [x] thread_cooperation, see #7836 
- [x] thread_exit, #7936
- [x] thread_flags, #7936
- [x] thread_flags_xtimer, #7936
- [x] thread_msg, see #7841 
- [x] thread_msg_block_w_queue, see #6720
- [x] thread_msg_block_wo_queue, see #6720
- [x] thread_msg_seq, #7936

Need some (trivial) modifications before being able to automate:
 - [x] pkg_u8g2 (#8039)
 - [x] posix_sleep (#7940) 
 - [x] trickle #7912

If we work together on this, we should be able to finish this list pretty quickly :-)

### PR DESCRIPTION
**Pull request by aabadie**
Partially adresses #7871.

This one was marked as 'needs trivial modifications' but in fact it works out of the box. @haukepetersen what kind of modifications are you thinking of ?

Tested on native and samr21-xpro.

### CODE REVIEW

File: tests/pkg_u8g2/tests/01-run.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+# Copyright (C) 2016 Mathias Tausig <mathias.tausig@fh-campuswien.ac.at>
```
Comment (basilfx) [inline tests/pkg_u8g2/tests/01-run.py]: Don't you want to include yourself?

Comment (aabadie) [reply tests/pkg_u8g2/tests/01-run.py]: Ah, yes copy-paste, will update

**Comment by basilfx** [comment]
Tested ACK.

Just for the record: since Python 3 is used, the default encoding of files is UTF-8, which should not lead to problems for the block characters.

**Comment by cladmi** [comment]
Why is it waiting for other PRs ?

**Comment by aabadie** [comment]
> Why is it waiting for other PRs ?

I wanted it to close #7871 when merged but apparently it has been reviewed faster then expected. Will change that.

**Comment by aabadie** [comment]
All green, merging !

**Pull request merged**